todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java

```diff
@@ -91,34 +108,31 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         FloatingActionButton fab =
                 (FloatingActionButton) getActivity().findViewById(R.id.fab_edit_task);
 
-        fab.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                mPresenter.editTask();
-            }
-        });
+        // RxBindings
+        mEditTaskClick = RxView.clicks(fab);
+        mCheckedChanges = RxCompoundButton
+                .checkedChanges(mDetailCompleteStatus)
+                .doOnNext(checked -> Log.i("TEST", "Pre-UpdatingFilter " + checked))
+                .filter(checked -> !mUpdating)
+                .share();
 
         return root;
     }
 
     @Override
-    public void setPresenter(@NonNull TaskDetailContract.Presenter presenter) {
+    public void setPresenter(@NonNull BasePresenter presenter) {
         mPresenter = checkNotNull(presenter);
     }
 
     @Override
-    public boolean onOptionsItemSelected(MenuItem item) {
-        switch (item.getItemId()) {
-            case R.id.menu_delete:
-                mPresenter.deleteTask();
-                return true;
-        }
-        return false;
-    }
-
-    @Override
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         inflater.inflate(R.menu.taskdetail_fragment_menu, menu);
+
+        // By using RxRelay, we can abstract Fragment's lifecycle implementation details.
```

Comment:
indeed. however in a similar manner you my apply this approach to all the other Observables and call subscribe() before even onCreate() is called.
Subject/Relays subscriptions however needs to be handled as well which is not done now.
another option is that there could be multiple subscribe() calls for example.